com.unity.netcode.gameobjects/Tests/Runtime/Metrics/NetworkObjectMetricsTests.cs

```diff
@@ -13,7 +13,9 @@ namespace Unity.Netcode.RuntimeTests.Metrics
 {
     internal class NetworkObjectMetricsTests : SingleClientMetricTestBase
     {
-        private const string k_NewNetworkObjectName = "TestNetworkObjectToSpawn";
+        // Keep less than 23 chars to avoid issues if compared against a 32-byte fixed string
+        //     since it will have "(Clone)" appended
+        private const string k_NewNetworkObjectName = "TestObjectToSpawn";
```

Comment:
does this 32-byte fixed string limit is also something to be afraid of at runtime for other objects?
for instance, if a user were to name their gameobject with a name longer than 32 chars, would that suddenly break metrics or is this issue unique to this test?

Reply:
No it shouldn't - it is specifically the equality check that's causing the issue, and this equality check is only done in this test. In the PR that changes over to fixed strings, I have sanity checks in place to avoid the truncation errors and also tests to verify that

Reply:
Also, the length that I'm moving to is 64 bytes. It seems to be a bug that it selects 32 bytes when doing the equality check. There's no overload for `FixedString64Bytes == string` and so it implicitly converts `string` to one of the parameters in some of the overloaded `==` operators. Although `FixedString64Bytes == FixedString64Bytes` exists, for some reason the compiler is choosing `FixedString64Bytes == FixedString32Bytes`.

The real fix here should be that the `string` is explicitly cast to `FixedString64Bytes`. However, that would introduce a backwards incompatibility with tools packages. This change is less ideal, but not ugly, and works with both the old and new versions of tools package.